server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/FilterAllocationDecider.java

```diff
@@ -81,6 +81,11 @@
         Setting.prefixKeySetting(CLUSTER_ROUTING_EXCLUDE_GROUP_PREFIX + ".", (key) ->
             Setting.simpleString(key, (value, map) -> IP_VALIDATOR.accept(key, value), Property.Dynamic, Property.NodeScope));
 
+    public static final boolean CLUSTER_ROUTING_REQUIRE_GROUP_SETTING_OMIT_DEFAULTS = true;
```

Comment:
do we need these constants, they are only used in one place? Can we please just use the raw boolean?

Reply:
Yes, I'll inline them as they aren't used in tests anymore, thank you.